sshd-core/src/main/java/org/apache/sshd/server/auth/pubkey/UserAuthPublicKey.java

```diff
@@ -176,7 +180,7 @@ public Boolean doAuth(Buffer buffer, boolean init) throws Exception {
                     username, session, alg, KeyUtils.getFingerPrint(key));
         }
 
-        return Boolean.TRUE;
+        return key instanceof OpenSshCertificate;
```

Comment:
This is wrong. Change back to `return Boolean.TRUE`.  The method is supposed to return TRUE if authentication was successful, FALSE if unsuccessful, and null if not complete yet.